### PR TITLE
Fix AgnocompleteWidgetMixin for template-based widgets (Django>=1.11)

### PR DESCRIPTION
**Pull request by joehybird**

### CODE REVIEW

**Comment by brunobord** [comment]
sorry, my latest merge introduces a conflict.

File: agnocomplete/widgets.py [hunk 1]
```diff
@@ -79,6 +78,34 @@ def build_attrs(self, base_attrs, extra_attrs=None):
...
+        """
+        def _agnocomplete_selected_options(self, options, value):
+            selected_options = {}
+
+            for opt in options:
+                opt_value = text(opt.get('value'))
+                opt_selected = (opt_value in value)
+
+                opt['selected'] = opt_selected
+                opt['attrs'][u'selected'] = opt_selected
```
Comment (wo0dyn) [inline agnocomplete/widgets.py]: Nitpick: if we put `from __future__ import unicode_literals` at the top of the file, do you think we can get a ride of `u'selected' here?

File: demo/tests/test_fields.py [hunk 2]
```diff
@@ -82,7 +143,21 @@ def test_empty(self):
...
+                                                        to_field_name='email')
+
+    def setUp(self):
+        super(MultipleModelSelectTest, self).setUp()
+        self.alice = Person.objects.get(pk=1)
+        self.bob = Person.objects.get(pk=3)
+
+        self.aliceinchains = Person.objects.get(pk=2)
+        self.aliceinchains.email = self.alice.email
+        self.aliceinchains.save()
```
Comment (wo0dyn) [inline demo/tests/test_fields.py:158]: 🤘

**Review by wo0dyn — APPROVED** [review]
I didn't spot any issue. It looks really good to me!
Thanks @joehybird!

File: agnocomplete/widgets.py (hunk 1 above)
Comment (brunobord) [reply agnocomplete/widgets.py]: I agree.

**Comment by brunobord** [comment]
please rebase from master and amend changelog accordingly, and we'll be fine :o)

**Comment by wo0dyn** [comment]
Thanks for the contribution @joehybird!

**Pull request merged**